### PR TITLE
[qt5cpp] delete callback data allocated before signal emission

### PR DESCRIPTION
**Pull request by etherealjoy**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [X] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR
Before emitting the signals declared in the API class when the callback from HTTP Worker object is received, there is memory allocation being made which is unknown to user code.

This PR adds a wrapper to any memory allocation being done and schedule deleteLater so that once the signal callback has been invoked, the object can be scheduled to be deleted.
Ideal case would be all memory allocation done by the generated code will be deallocated inside it, and user allocated memory will be done on user code.

Tests
- [X] PetStore 
- [X] Docker API tests with Valgrind

There is a limitation with the current PetStore tests because when the callback is received the loop is exited which prevents deleteLater from running. So the PetStore still have the memory Leak visible.

@stkrwork @fvarose @ravinikam

### CODE REVIEW

**Comment by etherealjoy** [comment]
@wing328 How do I remove this annoying old baseline which keeps popping up even after I have resync my fork with upstream/master?

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-header.mustache [hunk 1]
```diff
@@ -21,6 +22,19 @@ namespace {{this}} {
     QString stringValue(qint64 value);
     QString stringValue(bool value);
 
+    template <typename ObjPtrT>
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-header.mustache]: maybe move this class to its own file

Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-header.mustache]: Do you want to suggest a more appropriate class name also?

Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-header.mustache]: OK, I came up with object wrapper which is what it is..., :)

File: modules/swagger-codegen/src/main/resources/qt5cpp/QObjectWrapper.h.mustache [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#define {{prefix}}_QOBJECT_WRAPPER_H
+
+#include <QObject>
+
+{{#cppNamespaceDeclarations}}
+namespace {{this}} {
+{{/cppNamespaceDeclarations}}
+
+    template <typename ObjectPtrT>
+    class QObjectWrapper : public QObject {
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/QObjectWrapper.h.mustache]: Maybe also add the prefix to this class name as well

**Pull request merged**